### ISSUE TITLE
Feature Request: Take saplings from Inventory

### ISSUE THREAD

**Issue by roberestarkk** [ThizThizzyDizzy/tree-feller#36]
Would be fantastic if there were a setting to enable taking saplings from the inventory to replant in the case where the leaves being broken don't drop a sapling.

Birch trees in particular I've discovered do not always drop a sapling, but they do on average drop more than they don't, so I find I have plenty of spares in my inventory, but trees are not being replanted sometimes because the tree itself doesn't drop anything.

Probably taking the first found sapling from the inventory would work most consistently, but you could also decide to search for the sapling that corresponds with the wood/leaf type as an alternative.

**Comment by ThizThizzyDizzy** [comment]
Ooh, I never even thought of that, thanks for the suggestion :P

**Comment by roberestarkk** [comment]
I didn't realise there was a build for this, sorry for the late reply.

How does it work?
I was expecting it to be another mode alongside spawning, or for it to work with spawn = 0.

But if I set
`Replant = Yes`
`Use Inventory Saplings = Yes`
`Spawn to replant = No, only normal drops`

It doesn't replant if the tree doesn't drop any saplings, even if I've got them in my inventory

If I use `Spawn to replant = Yes, if it doesn't drop any`
It still doesn't replant if the tree doesn't drop any saplings, even if I've got them in my inventory

If I use `Spawn to replant = Yes, always`
It at least places a sapling all the time, but it doesn't consume from inventory either.

.

Also, as an aside, is it Shift+Click to save the config, or is it 'Crouch'+Click but mostly people don't rebind it even though Ctrl is crouch in all the other games and the Minecraft default keybinds are baffling?

**Comment by roberestarkk** [comment]
Turns out it's throwing an error, I probably should have checked for that in hindsight but I was excited to see a new build

```
[21:03:19] [Server thread/WARN]: [TreeFeller] Task #1093 for TreeFeller v1.17.1 generated an exception
         java.lang.NullPointerException: Cannot invoke "org.bukkit.inventory.ItemStack.getType()" because "stack" is null
         at com.thizthizzydizzy.treefeller.Sapling.tryPlace(Sapling.java:58) ~[TreeFeller-1.17.1.jar:?]
         at com.thizthizzydizzy.treefeller.Sapling.tick(Sapling.java:36) ~[TreeFeller-1.17.1.jar:?]
         at com.thizthizzydizzy.treefeller.TreeFeller$3.run(TreeFeller.java:1402) ~[TreeFeller-1.17.1.jar:?]
         at org.bukkit.craftbukkit.v1_18_R1.scheduler.CraftTask.run(CraftTask.java:101) ~[paper-1.18.1.jar:git-Paper-169]
         at org.bukkit.craftbukkit.v1_18_R1.scheduler.CraftScheduler.mainThreadHeartbeat(CraftScheduler.java:483) ~[paper-1.18.1.jar:git-Paper-169]
         at net.minecraft.server.MinecraftServer.tickChildren(MinecraftServer.java:1547) ~[paper-1.18.1.jar:git-Paper-169]
         at net.minecraft.server.dedicated.DedicatedServer.tickChildren(DedicatedServer.java:480) ~[paper-1.18.1.jar:git-Paper-169]
         at net.minecraft.server.MinecraftServer.tickServer(MinecraftServer.java:1470) ~[paper-1.18.1.jar:git-Paper-169]
         at net.minecraft.server.MinecraftServer.runServer(MinecraftServer.java:1264) ~[paper-1.18.1.jar:git-Paper-169]
         at net.minecraft.server.MinecraftServer.lambda$spin$0(MinecraftServer.java:317) ~[paper-1.18.1.jar:git-Paper-169]
         at java.lang.Thread.run(Thread.java:833) ~[?:?]
```

Which is thrown multiple times per second and continues to do so until the server restarts, even after I `/treefeller off` and `/treefeller reload`
It doesn't happen if the `use_inventory_saplings` setting is true

**Comment by ThizThizzyDizzy** [comment]
Originally added in v1.17 (forgot to close the issues .-.)
Fixed in v1.17.2
